src/Util/voxelutility.cpp

```diff
@@ -0,0 +1,78 @@
+#include <fstream>
+#include <iostream>
+
+#include "voxelutility.hpp"
+
+int VoxelUtility::from_qb(const char * path,
+            std::vector<uint32_t>& data,
+            std::vector<int>& center,
+            std::vector<int>& size){
+    //Read the tiles from the path specified, and save
+
+    //This is partially from goxel with GPL license
+    std::cout << "Reading form " << path << "\n";
+    FILE *file;
+    //int x, y, z; // position in terrain
+    uint32_t void_, compression; // void int 32 used to read 32 bites without saving it.
+    int32_t x_center, y_center, z_center;
+    uint8_t v[4];
+
+    file = fopen(path, "rb");
+    READ<uint32_t>(void_, file); // version
+    //std::cout << void_ << std::endl;
+    READ<uint32_t>(void_, file);   // color format RGBA
+    //std::cout << void_ << std::endl;(void) void_;
+    READ<uint32_t>(void_, file);   // orientation right handed // c
+    //std::cout << void_ << std::endl;(void) void_;
+    READ<uint32_t>(compression, file);   // no compression
+    //std::cout << void_ << std::endl;(void) void_;
+    READ<uint32_t>(void_, file);   // vmask
+    //std::cout << void_ << std::endl;(void) void_;
+    READ<uint32_t>(void_, file);
+    //std::cout << void_ << std::endl;(void) void_;
+    // none of these are used
+
+    int8_t name_len;
+    READ<int8_t>(name_len, file);
+    std::cout << "name length: " << (int) name_len << std::endl;
+    char* name = (char*) malloc (name_len);
```

Comment:
NEVER USE MALLOC/FREE IN C++ USE NEW/DELETE
AND NEVER USE NEW/FREE WHEN YOU CAN USE THE STL

USE `std::string`